### PR TITLE
Emit an error when runned on Linux without GUI

### PR DESCRIPTION
**Pull request by wentwrong**
## Purpose
Emit an error when trying to run tests on Linux without the graphics subsystem.

## Approach
Check that ```DISPLAY``` environment variable is set. If it's not, then throw an error when trying to run tests on local and non-headless browser.

## References
DevExpress/testcafe#4461

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

**Comment by wentwrong** [comment]
@testcafe-build-bot retest

File: src/browser/provider/built-in/remote.js [hunk 1]
```diff
@@ -35,6 +35,13 @@ export default {
...
 
     async isLocalBrowser (browserId) {
+        // NOTE:
+        // if browserId is not specified, then a browser is not yet started
+        // we may assume that it's not local, because
+        // otherwise we'll just disable window manipulation function's after the browser will be started
+        if (!browserId)
+            return false;
+
         return this.localBrowsersFlags[browserId];
```
Comment (AndreyBelym) [inline src/browser/provider/built-in/remote.js:45]:
```suggestion
        return browserId && this.localBrowsersFlags[browserId];
```

File: src/browser/provider/plugin-host.js [hunk 2]
```diff
@@ -98,12 +98,17 @@ export default class BrowserProviderPluginHost {
     }
 
     // Extra functions
-    // NOTE: The browserName argument is optional, and must be supplied if the browserId argument is not valid (browser is not opened)
+    // NOTE:
+    // The browserName argument is optional, and must be supplied if the browserId argument is not valid
```
Comment (AndreyBelym) [inline src/browser/provider/plugin-host.js:102]: 👍 for the comment

File: src/errors/runtime/templates.js [hunk 3]
```diff
@@ -102,5 +102,9 @@ export default {
     [RUNTIME_ERRORS.unexpectedIPCBodyPacket]:                            'Cannot create an IPC message due to an unexpected IPC body packet.',
     [RUNTIME_ERRORS.unexpectedIPCTailPacket]:                            'Cannot create an IPC message due to an unexpected IPC tail packet.',
     [RUNTIME_ERRORS.cannotUseAllowMultipleWindowsOptionForLegacyTests]:  'You cannot run Legacy API tests in multi-window mode.',
-    [RUNTIME_ERRORS.cannotUseAllowMultipleWindowsOptionForSomeBrowsers]: 'You cannot use multi-window mode in {browsers}.'
+    [RUNTIME_ERRORS.cannotUseAllowMultipleWindowsOptionForSomeBrowsers]: 'You cannot use multi-window mode in {browsers}.',
+    [RUNTIME_ERRORS.cannotRunLocalNonHeadlessBrowserWithoutDisplay]:     'You run {browserAlias} browser with graphic interface in Linux without graphic subsystem. ' +
```
Comment (AndreyBelym) [inline src/errors/runtime/templates.js:106]: AFAIR we discussed that we should suggest using `chrome:headless`, `chromium:headless` or `firefox:headless` in this case.

File: src/runner/bootstrapper.ts [hunk 4]
```diff
@@ -58,6 +59,7 @@ interface Filter {
 }
 
 interface BrowserInfo {
+    alias?: string;
```
Comment (AndreyBelym) [inline src/runner/bootstrapper.ts:62]: I feel it should be a regular property, not an optional one.

File: src/runner/bootstrapper.ts [hunk 5]
```diff
@@ -186,6 +188,24 @@ export default class Bootstrapper {
         RemoteBrowserProvider.canDetectLocalBrowsers = false;
     }
 
+    private static async _checkThatTestsCanRunWithoutDisplay (browserInfoSource: BrowserInfoSource[]): Promise<void> {
+        for (const browserInfo of browserInfoSource) {
+            if (browserInfo instanceof BrowserConnection)
```
Comment (AndreyBelym) [inline src/runner/bootstrapper.ts:193]: I still think that we can remove this condition. From the auth check function too, but it's better to do it in a separate PR.

File: src/errors/runtime/templates.js (hunk 3 above)
Comment (wentwrong) [reply src/errors/runtime/templates.js]: @VasilyStrelyaev, can you take a look at this warning message?

> You run "{browserAlias}" browser with graphic interface in Linux without graphic subsystem. Try to run in headless mode. For more information see https://devexpress.github.io/testcafe/documentation/guides/concepts/browsers.html#test-in-headless-mode

Comment (VasilyStrelyaev) [reply src/errors/runtime/templates.js]: What about:

```js
    [RUNTIME_ERRORS.cannotRunLocalNonHeadlessBrowserWithoutDisplay]:     'Your Linux version does not have a graphic subsystem to run {browserAlias} with a GUI. You can launch {browserAlias} in headless mode (use \'{browserAlias}:headless\' as the browser alias). For more information, see {link}.'
```

(checked with Dirk)

Comment (wentwrong) [reply src/errors/runtime/templates.js]: But there is a problem with this approach. When the user specify a browser by `path`:
```
testcafe "path:/usr/bin/chromium" test.js
```
`browserAlias` will be equal to `path:/usr/bin/firefox` and the following error message will appear:

> Your Linux version does not have a graphic subsystem to run path:/usr/bin/firefox with a GUI. You can launch path:/usr/bin/firefox in headless mode (use **'path:/usr/bin/firefox:headless'** as the browser alias). For more information, see https://devexpress.github.io/testcafe/documentation/guides/concepts/browsers.html#test-in-headless-mode'

As described in the link above, the user should use the browser alias instead of the `path:` prefix. For example:
```
testcafe "firefox:/usr/bin/firefox:headless" test.js
```

Thus, just as a suggestion, I propose:

> Your Linux version does not have a graphic subsystem to run '{browserAlias}' with a GUI. You can launch the browser in headless mode. If you use a portable version of the browser, then use the browser alias instead of the 'path' prefix. For more information, see {link}.

By the way, are there any style-guides for using `'` instead of `"`? (may look inconsistent).

File: test/functional/setup.js [hunk 6]
```diff
@@ -181,6 +183,12 @@ before(function () {
             return openRemoteBrowsers();
         })
         .then(() => {
+            if (config.useLocalBrowsers && !config.useHeadlessBrowsers && os.linux && !detectDisplay())
```
Comment (AndreyBelym) [inline test/functional/setup.js:186]: I'm afraid we can't do this thing, since even if your test doesn't require the browser specified in the task configuration, it starts together with all other functional tests that rely on that browser.

Comment (wentwrong) [reply test/functional/setup.js]: Shame on me. I didn't notice that the functional tests (in local and non-headless browsers) are running in Travis with enabled `xvfb` service.

File: src/errors/runtime/templates.js (hunk 3 above)
Comment (VasilyStrelyaev) [reply src/errors/runtime/templates.js]: Checked with Dirk v2:

```js
Your Linux version does not have a graphic subsystem to run {browserAlias} with a GUI. You can launch the browser in headless mode. If you use a portable browser version, specify the browser alias before the path instead of the 'path' prefix. For more information, see {link}.
```

Comment (VasilyStrelyaev) [reply src/errors/runtime/templates.js]: I always use single quotes, but probably we have double quotes somewhere. IDK about the standard though, I should ask I guess.
